### DISCUSSION TITLE
How to get/set output filename from within a plugin?

### DISCUSSION THREAD

**Discussion by b-hawk** [streamlink/streamlink#4737] [Q&A] [answered]
How can I get (and then maybe change) the output filename specified in the streamlink options (-o) from within a plugin?

**Comment by bastimeyer** [comment, marked as answer]
CLI parameters are part of `streamlink_cli`, not `streamlink`. The `--output`/`-o` parameter is telling `streamlink_cli` what to do with the returned and selected `Stream` instance of a plugin, namely writing its data to the filesystem, and not to the spawned player process's `stdin` pipe, or elsewhere.

Other CLI parameter values are [set on the streamlink session](https://streamlink.github.io/api.html#streamlink.Streamlink.set_option) by streamlink_cli, like `http-headers` for example, which are then available in plugins, stream implementations or anywhere else where the session instance is available.

So no, you can't change `streamlink_cli` related user inputs via plugins.
